### ISSUE TITLE
Support reading environment variables passed to skaffold from file (e.g. .env)

### ISSUE THREAD

**Issue by stanislav-zaprudskiy** [GoogleContainerTools/skaffold#4106]
Considering `skaffold` could use environment variables to populate templated fields - https://skaffold.dev/docs/environment/templating/, it would super convenient to have the ability to pass environment variables to skaffold from file - similar to (or the same as) how [`docker-compose` does that](https://docs.docker.com/compose/environment-variables/).

Originally, this feature request just got lost here - https://github.com/GoogleContainerTools/skaffold/issues/543.

**Comment by balopat** [comment]
We also had a PR around this: https://github.com/GoogleContainerTools/skaffold/pull/1458

**Comment by limbuu** [comment]
+1

**Comment by nkubala** [comment]
when we shut down https://github.com/GoogleContainerTools/skaffold/pull/1458 last year, I think we were a bit closed-minded on our stance to templating as a whole, in that we were strictly against it in skaffold. over time, I think our opinion on this has relaxed: there are scenarios in which introducing environment variables into the skaffold pipeline can still fit in with our principles on the tool, and I think this issue is one of them. we don't want to allow full-blown templating of the skaffold.yaml and k8s resources (e.g. Jinja). however, we recognize that using env vars in build and deploy configuration can be a common use case, especially in CI/CD workflows, and we don't want to shut these use cases down completely.

tl;dr - our stance on supporting this feature in skaffold has changed, and while we probably won't prioritize it ourselves, we'd be open to a contribution if anyone is interested.

**Comment by simonoff** [comment]
Interesting.... we need such feature too!

**Comment by gsquared94** [comment]
This has been implemented. See https://skaffold.dev/docs/environment/env-file/

**Comment by mstaicu** [comment]
I've tried to specify environment variables in skaffold.env and then in my deployment yaml files expose those environment variables to application web servers running in containers but it didn't work. Can anyone explain what is the purpose of skaffold.env or if what I'm trying to achieve is intended by design?
